__tests__/integration/hathorwallet_others.test.js

```diff
@@ -0,0 +1,1247 @@
+   * 2 - getUtxos()
+   * 3 - getAllUtxos()
+   * 4 - getUtxosForAmount()
+   */
+
+  it('should mark utxos as selected', async () => {
+    // Validating utxo current state as not selected
+    let rawOutput = hWallet.getTx(txHash).outputs[oIndex];
+    loggers.test.log('Output before', rawOutput);
+    expect(rawOutput.selected_as_input).toStrictEqual(false);
+
+    // Marking it as selected
+    hWallet.markUtxoSelected(txHash, oIndex, true);
+
+    // Validation 1
+    rawOutput = hWallet.getTx(txHash).outputs[oIndex];
+    loggers.test.log('Output after', rawOutput);
+    expect(rawOutput.selected_as_input).toStrictEqual(true);
+
+    // Validation 2
+    expect(hWallet.getUtxos()).toMatchObject({
+      total_utxos_available: 0,
+      utxos: [expect.objectContaining({
+        tx_id: txHash,
+        locked: true,
+      })],
+    });
+
+    // Validation 3
+    const utxosGenerator = hWallet.getAllUtxos();
+    expect(utxosGenerator.next()).toStrictEqual({
+      value: undefined,
+      done: true,
+    });
+
+    // Validation 4
+    expect(() => hWallet.getUtxosForAmount(10)).toThrow('utxos to fill');
+  });
+
+  it('should mark utxos as not selected', async () => {
+    // Validating utxo current state as selected
+    let rawOutput = hWallet.getTx(txHash).outputs[oIndex];
+    expect(rawOutput.selected_as_input).toStrictEqual(true);
+
+    // Marking it as not selected
+    hWallet.markUtxoSelected(txHash, oIndex, false);
+
+    // Validation 1
+    rawOutput = hWallet.getTx(txHash).outputs[oIndex];
+    expect(rawOutput.selected_as_input).toStrictEqual(false);
+
+    // Validation 2
+    expect(hWallet.getUtxos()).toMatchObject({
+      total_utxos_available: 1,
+      utxos: [expect.objectContaining({
+        tx_id: txHash,
+        locked: false
+      })],
+    });
+
+    // Validation 3
+    const utxosGenerator = hWallet.getAllUtxos();
+    expect(utxosGenerator.next()).toStrictEqual({
+      value: expect.objectContaining({ txId: txHash }),
+      done: false,
+    });
+
+    // Validation 4
+    expect(hWallet.getUtxosForAmount(10)).toStrictEqual({
+      changeAmount: 0,
+      utxos: [expect.objectContaining({ txId: txHash })]
+    });
+  });
+});
+
+describe('consolidateUtxos', () => {
+  /** @type HathorWallet */
+  let hWallet1;
+  /** @type HathorWallet */
+  let hWallet2;
+  /** @type string */
+  let tokenHash;
+
+  /*
+   * The test initialization creates two wallets,
+   * Wallet1: Empty, with no transactions
+   * Wallet2: Containing HTR and custom token funds
+   */
+  beforeAll(async () => {
+    hWallet1 = await generateWalletHelper();
+    hWallet2 = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(hWallet2.getAddressAtIndex(0), 101);
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet2,
+      'Consolidate Token',
+      'CTK',
+      100,
+      { address: hWallet2.getAddressAtIndex(0) }
+    );
+    tokenHash = tokenUid;
```

Comment:
```suggestion
    { hash: tokenHash } = await createTokenHelper(
      hWallet2,
      'Consolidate Token',
      'CTK',
      100,
      { address: hWallet2.getAddressAtIndex(0) }
    );
```

Reply:
Here I first get the `tokenUid` of this transaction from the method results and then I update a variable on the `describe` scope called `tokenHash`.

I tried to make this declaration / assignment more concise, but the best I could think about (aside from what's in the code) was:
```js
const tokenTx = await createTokenHelper(...);
tokenHash = tokenTx.hash;
```

Do you think this would make the code's intentions clearer?

Reply:
My suggestion is that if you use deconstruction to instantiate the local variable, you could use it to update the `tokenHash` variable as well.
But I guess it would make it less clear so I don't think we should change it.